### ISSUE TITLE
Issues with equality types inside of structures

### ISSUE THREAD

**Issue by qtgeo1248** [azdavis/millet#56]
## Environment

- Millet version: 0.14.3
- Editor version: VS Code 1.88.1 (Universal)
- OS version: macOS 14.3.1

## Steps to reproduce

This is a shorted version of 150's testing library signature:
```sml
signature TEST =
sig
  (* `general_eq (name, x, y)` passes the test if `x = y`, fails otherwise. *)
  val general_eq : string * ''a * ''a -> unit
end
```

along with the structure
```sml
structure Test :> TEST =
struct
  fun general_eq (name, expected, actual) =
    (if expected = actual
    then ()
    else raise Fail name
    ) handle _ => raise Fail name
end
```

Type-annotating this modified structure:
```sml
structure Test' :> TEST =
struct
  fun general_eq (name, expected : ''a, actual : ''a) =
    (if expected = actual
    then ()
    else raise Fail name
    ) handle _ => raise Fail name
end
```
still gives the same behaviors outlined below

## Expected behavior

I think everything should work fine. Everything compiles as expected. I don't know enough about equality types to know if there are issues here.

## Actual behavior

I get this from Millet:
```
incompatible types: not an equality type because it contains a fixed non-equality type variable: `??a`
  expected `string * ??a * ??a -> unit`
     found `string * _ * _ -> unit` Millet[5006](https://github.com/azdavis/millet/blob/main/docs/diagnostics/5006.md)
5006.md(1, 1): click the blue '5006' for more info and/or advice for how to fix. in VS Code, set `millet.server.diagnostics.moreInfoHint.enable` to `false` to disable this hint.
```

**Comment by azdavis** [comment]
thanks for report. fixed in 399058b3aa6729ce3005dc5a273a77fb4f58dcf4

**Comment by azdavis** [comment]
fixed released in v0.14.4
